nbgitpuller/__init__.py

```diff
@@ -4,6 +4,12 @@
 from notebook.utils import url_path_join
 from tornado.web import StaticFileHandler
 import os
+import nest_asyncio
+
+# this allows us to nest usage of the event_loop from asyncio
+# being used by tornado in jupyter distro
+# Ref: https://medium.com/@vyshali.enukonda/how-to-get-around-runtimeerror-this-event-loop-is-already-running-3f26f67e762e
+nest_asyncio.apply()
```

Comment:
This monkeypatches the asyncio event loop, is this event loop shared across the whole of jupyter-server? If so are there any potential issues that might be caused in jupyter-server, extensions, or kernels that we should be aware of?

Reply:
@consideRatio @manics Erik do you have any sense of this? I completely based this solution on the reference I linked to in the comments. He seemed to be trying to solve a similar nested event loop issue in Jupyter.

I read a bunch more pieces. 
- Here is a [discussion](https://github.com/jupyter/notebook/issues/3397#issuecomment-419386811) about where the error occurs(Runtime error related to trying to start an already running loop); the ipykernel seems to be the issue. The nested_asyncio solution becomes a part of the thread in [August 2020](https://github.com/jupyter/notebook/issues/3397#issuecomment-675455123).
- Here is another from the [jupyter blog](https://blog.jupyter.org/ipython-7-0-async-repl-a35ce050f7f7) in 2018; at this point, they don't want to integrate the nested_asyncio package until there is more testing.
- I also tried upgrading the tornado and ipykernel packages; a random comment seemed to indicate it would solve the problem; after upgrading tornado to version 6.1(no change in my environment) and ipykernel to 6.6.1(up from 6.6.0) -- it didn't work. I still need the nested_asyncio package.
> pip3 install --upgrade tornado
> pip3 install --upgrade ipykernel

Reply:
I moved the nest_asyncio calls to the nbgitpuller-downloader-plugins themselves as well as added notes to the documentation that the import and call to nest_asyncio need to be included in any nbgitpuller-downloader-plugin.